### PR TITLE
feat: CFD-312 Create purchase methods in global settings

### PR DESCRIPTION
**Pull request by jacsamell**
# Description

-   Manage purchase methods in global settings

# Testing instructions

-   Create some purchase methods in global setting

# Type of change

-   [X] feat - A new feature
-   [ ] fix - A bug fix
-   [ ] docs - Documentation only changes
-   [ ] codestyle - Changes that do not affect the meaning of the code (white-space, formatting, missing semi-colons, etc)
-   [ ] refactor - A code change that neither fixes a bug nor adds a feature
-   [ ] perf - A code change that improves performance
-   [ ] test - Adding missing tests or correcting existing tests
-   [ ] build - Changes that affect the build system or external dependencies (example scopes: gulp, broccoli, npm)
-   [ ] ci - Changes to our CI configuration files and scripts (example scopes: Travis, Circle, BrowserStack, SauceLabs)
-   [ ] chore - Other changes that don't modify src or test files
-   [ ] revert - Reverts a previous commit
-   [ ] content - Changes to content or copy

### CODE REVIEW

File: repos/fdbt-site/src/pages/api/managePurchaseMethod.ts [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+        };
+
+        if (errors.length > 0) {
+            updateSessionAttribute(req, GS_PURCHASE_METHOD_ATTRIBUTE, { inputs: salesOfferPackage, errors });
+
+            redirectTo(res, '/managePurchaseMethod');
+            return;
+        }
+
+        await insertSalesOfferPackage(getAndValidateNoc(req, res), salesOfferPackage);
```
Comment (matt-twomey) [inline repos/fdbt-site/src/pages/api/managePurchaseMethod.ts:75]: Do we need to check that we don't already have a purchase method with the same name?

File: repos/fdbt-site/src/utils/index.ts [hunk 2]
```diff
@@ -220,3 +220,11 @@ export const checkIfMultipleOperators = (ctx: NextPageContextWithSession): boole
 
 export const getCsrfToken = (ctx: DocumentContextWithSession | NextPageContextWithSession | NextPageContext): string =>
     (ctx.res as ResponseWithLocals)?.locals?.csrfToken ?? '';
+
+export const toArray = (thing: string | string[] | undefined): string[] => {
```
Comment (matt-twomey) [inline repos/fdbt-site/src/utils/index.ts:224]: I think this is repeated in `repos/fdbt-site/src/pages/api/salesOfferPackages.ts` - shall we import from utils in this location too?

File: repos/fdbt-site/tests/pages/api/managePurchaseMethod.test.ts [hunk 3]
```diff
@@ -0,0 +1,240 @@
...
+        );
+        expect(res.writeHead).toBeCalledWith(302, {
+            Location: '/managePurchaseMethod',
+        });
+    });
+
+    it('redirects back to /managePurchaseMethod if no name', () => {
+        const { req, res } = getMockRequestAndResponse({
+            body: {
+                name: 'my name',
```
Comment (matt-twomey) [inline repos/fdbt-site/tests/pages/api/managePurchaseMethod.test.ts:183]: Should name be omitted here, and paymentMethods included?

Comment (jacsamell) [reply repos/fdbt-site/tests/pages/api/managePurchaseMethod.test.ts:183]: Good spot 👍

File: repos/fdbt-site/src/utils/index.ts (hunk 2 above)
Comment (jacsamell) [reply repos/fdbt-site/src/utils/index.ts:224]: Good idea 👍

File: repos/fdbt-site/src/pages/api/managePurchaseMethod.ts (hunk 1 above)
Comment (jacsamell) [reply repos/fdbt-site/src/pages/api/managePurchaseMethod.ts:75]: Good spot 👍

**Review by matt-twomey — APPROVED** [review]

**Pull request merged**